service/src/ConstConfigIOGroup.cpp

```diff
@@ -0,0 +1,424 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "ConstConfigIOGroup.hpp"
+
+#include <cmath>
+
+#include <iostream>
+#include <iterator>
+#include <sstream>
+
+#include "geopm/Helper.hpp"
+#include "geopm/PlatformTopo.hpp"
+#include "geopm/Exception.hpp"
+#include "geopm/Agg.hpp"
+#include "geopm_topo.h"
+#include "config.h"
+
+using json11::Json;
+
+namespace geopm
+{
+    const std::string ConstConfigIOGroup::M_PLUGIN_NAME = "CONST_CONFIG";
+    const std::string ConstConfigIOGroup::M_SIGNAL_PREFIX = M_PLUGIN_NAME + "::";
+    const std::map<std::string, ConstConfigIOGroup::m_signal_desc_s>
+        ConstConfigIOGroup::M_SIGNAL_FIELDS = {
+            {"description", {Json::STRING, true}},
+            {"units", {Json::STRING, true}},
+            {"domain", {Json::STRING, true}},
+            {"aggregation", {Json::STRING, true}},
+            {"values", {Json::ARRAY, true}}
+        };
+
+    ConstConfigIOGroup::ConstConfigIOGroup()
+    {
```

Comment:
I thought this would look something like:
```suggestion
    ConstConfigIOGroup::ConstConfigIOGroup()
        : ConstConfigIOGroup(M_DEFAULT_CONFIG_FILE)
    {
```

Reply:
Addressing both (this and Daniel's) comments here:

The feature description calls for accepting a config file path via an environment variable. So the default constructor (called by the ``make_plugin()`` method) will read the config file (perhaps a default path could be used if none provided) and then proceed with parsing. I provided the specialized constructor as a convenience for the unit tests. I might refactor this a bit if I can think of a better solution.

Reply:
I think we need to refactor the constructor a bit.  There are some code paths that we cannot test in CI the way it is.  I put a sketch together with a few TODOs.  It's here:
https://github.com/bgeltz/geopm/commit/f51e3e84f4c8d5f19b0c86c6c8c4005b3b470a0c
https://github.com/bgeltz/geopm/tree/issue-2619

With a refactor like I suggested, this is easier to test and maintain.  Take a look and let me know if you have questions.  The unit tests should create real/on-disk input files that are parsed here.  Both paths should be tested (env specified file path and default path).

Reply:
@bgeltz Do you want me to write the unit tests that create real/on-disk input files?

Reply:
Yes